Adding a sample inventory and basic instructions for following a BYO …

etsauer opened the pull request:
…infra model

#### What does this PR do?
@oybed @logandonley as discussed last week, this is my first attempt to make the host prep/byo infra story a thing.

FYI, I am targeting the use of this as training material for our practice architect enablement session next week.

#### How should this be manually tested?
Read through the README, and follow links in the BYO Infra section to read further.

#### Is there a relevant Issue open for this?
n/a

#### Who would you like to review this?
cc: @redhat-cop/casl

bagnaram commented:
I like the use-case of breaking apart the documentation for the various infrastructures.

etsauer commented:
@logandonley @oybed did either of you get a chance to test this?

oybed commented on README.md:
```diff
@@ -3,24 +3,25 @@
 
+### Automation of Cluster Installation on Your Own (BYO) Infrastructure
+
+Additionally, for those who are using infrastructure that is either not yet fully automated, or not yet supported through a CASL provider, we provide a [Bring Your Own Infrastructure](./docs/BYO_INFRASTRUCTURE.adoc) guide to using CASL.
```
A bit "too much repeat" starting with the same word in this section as the previous one  - maybe just remove the beginning `Additionally`.

oybed commented on docs/BYO_INFRASTRUCTURE.adoc:
```diff
@@ -0,0 +1,60 @@
+While the link:../inventory/sample.byo.example.com.d/[sample.byo.example.com] sample inventory provides most of the variables you need to get your cluster prepped and installed, there are a few variables in the `sample.byo.example.com.d/inventory/group_vars/all.yml` file that need to be filled in to match your environment.
+
+- The `ansible_user: <your ssh key user>` variable must match the user you will use to log into the cluster machines
+- You must fill in the `rhsm_` variables to match your Red Hat Subscription Manager account. You can do this one of two ways:
```
May want to mention that `rhsm_register: True` has to be set to successfully register the hosts.

oybed commented on docs/BYO_INFRASTRUCTURE.adoc:
```diff
@@ -0,0 +1,60 @@
+rhsm_activationkey: 'casl-latest'
+----
+
+== Running the CASL Tools on my BYO infrastructure
```
Let's add `provisioning` - i.e.: "CASL Provisioning Tools"

oybed commented on docs/BYO_INFRASTRUCTURE.adoc:
```diff
@@ -0,0 +1,60 @@
+
+== Running the CASL Tools on my BYO infrastructure
+
+Once the infrastructure has been manually provisioned and your inventory is prepped, you should be ready to run the CASL tools.
```
add "provisioning" (as above)

oybed commented on inventory/sample.byo.example.com.d/inventory/group_vars/OSEv3.yml:
```diff
@@ -0,0 +1,50 @@
+deployment_type: openshift-enterprise
+openshift_deployment_type: openshift-enterprise
+openshift_master_cluster_method: native
+openshift_release: v3.7
```
Let's make it `v3.9`

oybed commented on inventory/sample.byo.example.com.d/inventory/group_vars/OSEv3.yml:
```diff
@@ -0,0 +1,50 @@
+
+# openshift_master_cluster_public_hostname must be something like:
+#   <prefix>.{{ env_id }}.{{ dns_domain }}
+openshift_master_cluster_public_hostname: master.{{ env_id }}.{{ dns_domain }}
```
the other sample inventories uses `console` - should we do the same here?

oybed commented on inventory/sample.byo.example.com.d/inventory/group_vars/OSEv3.yml:
```diff
@@ -0,0 +1,50 @@
+openshift_master_htpasswd_users:
+  admin: $apr1$7aiANAYb$TOUYVUqnBqBlD5AQEIMYw1
+
+openshift_hosted_router_selector: 'region=infra'
```
These aren't really needed to be set - any particular reason to include them? (if so, we should include them for the other sample inventories as well). Same for the registry below.

oybed commented on inventory/sample.byo.example.com.d/inventory/group_vars/OSEv3.yml:
```diff
@@ -0,0 +1,50 @@
+os_firewall_use_firewalld: true
+osm_cluster_network_cidr: 10.1.0.0/16
+
+openshift_enable_service_catalog: false
```
Should we just leave the "disabling" of the service catalog out?

oybed commented:
One other thought - with the latest doc changes, it really seems that CASL == "provisioning of OCP clusters" while in reality we are trying to make it be more than just that. Let's discuss.

oybed commented on docs/BYO_INFRASTRUCTURE.adoc:
```diff
@@ -0,0 +1,60 @@
+
+_Provision Infrastructure_ creates the servers/instances and any other supporting infrastructure required (network, DNS, load balancers, etc.). In general this phase
+  - Begins with having credentials with proper access to an Infrastructure provider.
+  - Ends with servers provisioned and running with an OS installed and SSH keys (with root or sudo) pushed.
```
May want to include some DNS info here + proper cert handling (especially when LBs are in-use)

etsauer commented:
@oybed addressed your comments. I chose to leave certificates out, as they are optional and I don't want to create too high of a barrier to start using the tools. Let me know what you think
